concordium-contracts-common/src/types.rs

```diff
@@ -1406,6 +1536,67 @@ pub struct Policy<Attributes> {
     pub items:             Attributes,
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_no_dup_kv_vec<A: quickcheck::Arbitrary + Ord, B: quickcheck::Arbitrary>(
+    g: &mut Gen,
+    size: usize,
+) -> Vec<(A, B)> {
+    use std::collections::BTreeMap;
+    let mut m: BTreeMap<A, B> = BTreeMap::new();
+    while m.len() < size {
+        let k = A::arbitrary(g);
+        let v = B::arbitrary(g);
+        m.insert(k, v);
+    }
+    m.into_iter().collect()
+}
+
+#[cfg(feature = "concordium-quickcheck")]
+const ATTRIBUTE_TAG_VEC_LENGTH: [u8; 15] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14];
+#[cfg(feature = "concordium-quickcheck")]
+fn gen_range_u64(g: &mut Gen, range: core::ops::Range<u64>) -> u64 {
+    let i: u64 = quickcheck::Arbitrary::arbitrary(g);
+    i % (range.end - range.start) + range.start
+}
+
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for OwnedPolicy {
+    //todo it is not clear to me if random instances of these are useful in tests
```

Comment:
TODO.
Not sure either, perhaps Ales has some input.

Reply:
I think it's an old comment. The current implementation should generate random, but valid instances.

Reply:
We should remove the todo. The instances are useful.